Cancelled VDJ / Late VDJ / ODB Pax Count adjustments

gnitsch opened the pull request.

gnitsch committed:
CancelledDVJ, LateDVJ, ODBPaxCount update

- added NeTEx facade to pseudo-model Deck Plan elements and Type Of Delays
- ODB Pax Count scope references' types adjusted
- Late DVJ adjustments
- Cancelled DVJ re-modelled based on Andrej's work

github-actions[bot] committed:
Lint and update documentation tables

gnitsch committed:
Merge branch 'main' into example_ticketbasedpassengercount

Aurige commented on xsd/opra_imports/opra_netex_facade.xsd:
```diff
@@ -0,0 +1,109 @@
+			</xsd:restriction>
+		</xsd:simpleContent>
+	</xsd:complexType>
+	<xsd:simpleType name="PassengerEntranceIdType">
```
There is already a PassengerEntranceIdType in netex_deckPlan_support.xsd ... it's always best to reusez already existing neTEx types
By the way, in XMLSpy there is a convenient search tool, that is worth using each time you need an element or type that may already exist in NeTEx or SIRI

gnitsch replied:
I believe it's not in the attached NeTEx subproject. So the opra_netex_facade.xsd serves as a temporary replacement for the NeTEx additions being developed, thus the OpRa parts can be thoroughly defined. The opra_netex_facade.xsd file will need to be removed from the final version.

Aurige replied:
it is ... you need to import netex_deckPlan_support.xsd

Aurige commented on xsd/opra_imports/opra_netex_facade.xsd:
```diff
@@ -0,0 +1,109 @@
+	<xsd:simpleType name="PassengerEntranceIdType">
+		<xsd:restriction base="netex:ObjectIdType"/>
+	</xsd:simpleType>
+	<xsd:complexType name="PassengerEntranceRefStructure">
```
same for PassengerEntranceRefStructure

gnitsch replied:
same as above

Aurige commented on xsd/opra_imports/opra_netex_facade.xsd:
```diff
@@ -0,0 +1,109 @@
+			</xsd:restriction>
+		</xsd:simpleContent>
+	</xsd:complexType>
+	<xsd:simpleType name="TypeOfDelayIdType">
+		<xsd:restriction base="netex:ObjectIdType"/>
+	</xsd:simpleType>
+	<xsd:complexType name="TypeOfDelayRefStructure">
+		<xsd:simpleContent>
+			<xsd:restriction base="netex:VersionOfObjectRefStructure">
+				<xsd:attribute name="ref" type="TypeOfDelayIdType" use="required">
+					<xsd:annotation>
+						<xsd:documentation>Identifier of referenced TYPE OF DELAY.</xsd:documentation>
+					</xsd:annotation>
+				</xsd:attribute>
+			</xsd:restriction>
+		</xsd:simpleContent>
+	</xsd:complexType>
```
to be compared and maybe aligned with SIRI **_DelaysStructure_** ...

gnitsch committed:
External Pax Count and Expected Pax Count introduction

- added external pax count
- added expected pax count

gnitsch committed:
Merge branch 'example_ticketbasedpassengercount' of https://github.com/OpRa-CEN/OpRa into example_ticketbasedpassengercount

github-actions[bot] committed:
Lint and update documentation tables

gnitsch committed:
added ExpectedPassengerCount example

- see the Sharepoint DOC for sample values
- ExpectedPassengerScopeGroup typo fix

gnitsch committed:
PT SCOPE fix

- was not saved before previous commit

github-actions[bot] committed:
Lint and update documentation tables

gnitsch committed:
Refactor and Dated Vehicle Journey examples

Refer to Sample Network doc on Sharepoint.
Changes:
- replaced elements with element references where NeTEx, SIRI, or OpRa definitions already exist
- changed opra namespace to https://transmodel-cen.eu/opra/1.0
- added CancelledAndLateDatedVehicleJourneys.xml example

gnitsch committed:
Merge branch 'example_ticketbasedpassengercount' of https://github.com/OpRa-CEN/OpRa into example_ticketbasedpassengercount

gnitsch committed:
placeholder.xml fix

github-actions[bot] committed:
Lint and update documentation tables

gnitsch committed:
VehicleLoadEntry replaced with SIRI occupancy references

- SIRI OccupancyScopeFilterGroup and OccupancyValuesGroup re-used
- annotations reviewed for CIB.

gnitsch committed:
Merge branch 'example_ticketbasedpassengercount' of https://github.com/OpRa-CEN/OpRa into example_ticketbasedpassengercount

gnitsch committed:
Re-introduced TimebandRef and fixed examples

- added TimebandRef to multiple passenger count groups.

github-actions[bot] committed:
Lint and update documentation tables

tim-rivett committed:
Updated PreparednessEnumeration

Added Statistics to the Preparedness enumeration as this is one of the stages of preparing data.

tim-rivett committed:
Revert "Updated PreparednessEnumeration"

This reverts commit bacb18e8aef463176d200a3ff100ce203296adc4.

gnitsch committed:
Added Request/Response objects, extended PT SCOPE

- request / response elements (with bulk requests capabilities)
- moved TimeBandRef and DayTypeRef to PT SCOPE
- added examples for requests and Late DVJ response

gnitsch committed:
Merge branch 'example_ticketbasedpassengercount' of https://github.com/OpRa-CEN/OpRa into example_ticketbasedpassengercount

gnitsch committed:
frames in OpraRequests fix

gnitsch committed:
Abstract service structures re-using existing SIRI elements

Added SIRI-like services:
- abstract OpRa element layer
- collections of abstract Opra elements
- replaced SIRI functional areas with abstract collections
- capability of handling bulk compound requests, returning bulk compound responses

Added services for use cases
Added CALENDAR AWARE PT SCOPE
Updated and validated examples
Added all NeTEx frames to imports

gnitsch committed:
Adding services, discovery

- PlannedService service
- ActualService service
- *DiscoveryRequest definitions for use cases
- *DiscoveryDelivery definitions for use cases

gnitsch committed:
addition to statistics

- added TotalNumberOfOccurrences

Aurige commented:
The validation issue is comming from the Validation script: it validate against OpRa_publication.xsd ( xmllint --noout --schema xsd/OpRa_publication.xsd "$file"; ) but this file is a request and need to be validate against Opra.xsd

Aurige commented:
I think that everything should be validate against opra.xsd (but the way OpRa.xsd would be better since we have OpRa_publication.xsd ... same for opra_request.xsd )

Further more the OpRa.xsd header doesn't feet to validate Deliveris .... :-(

it should be something like:
```
    <!-- ===SIRI system IDs for  request =========================================================== -->
	<xsd:import namespace="http://www.siri.org.uk/siri" schemaLocation="opra_imports/opra_siri_imports.xsd"/>
	<!-- ===Regular netex============================================================== -->
	<xsd:import namespace="http://www.netex.org.uk/netex" schemaLocation="opra_imports/opra_netex_imports.xsd"/>
	<!-- ===Includes (OpRa)============================================================== -->
	<xsd:include schemaLocation="opra_framework/opra_payload_framework.xsd"/>
    <xsd:include schemaLocation="OpRa_publication.xsd" />
    <xsd:include schemaLocation="opra_request.xsd" />
```

(I just added the  <xsd:include schemaLocation="OpRa_publication.xsd" />)

gnitsch commented:
Thanks. I'd prefer opra_publication.xsd, since everything is in lower camel case elsewhere.

Aurige commented:
> Thanks. I'd prefer opra_publication.xsd, since everything is in lower camel case elsewhere.

Ok, I let you do the fixes

gnitsch committed:
xml lint validation fix

gnitsch committed:
case tinkering

gnitsch committed:
removal of erroneous example

gnitsch committed:
removal of erroneous example

gnitsch committed:
annotations and examples reviewed

gnitsch committed:
CancelledDatedVehicleJourney cleanup

Aurige commented:
we still need to update de validation script now (    **xmllint --noout --schema xsd/opra.xsd "$file";**   and not     **xmllint --noout --schema xsd/OpRa_publication.xsd "$file";**  )
